### ISSUE TITLE
Boundary check for device data failed ZodError

### ISSUE THREAD

**Issue by frequentsailor** [lstrojny/homebridge-prometheus-exporter#7]
First of all thank you very much for making this plugin! I have been looking for something like it for some time now. I hope I filled everything correctly. If not please let me know.

**Describe The Bug:**
I installed the plugin and configured it according to the readme. However if I open the /metrics page it is blank. In the logs I found a very long list of errors. It starts of with "Boundary check for device data failed ZodError:". 

**To Reproduce:**
I am not sure what the problem is. I have quite a few sensors and devices in homebridge (82 devices total). And some of the names have special characters (German) in them.

**Expected behavior:**
No error messages in the log file and information on the /metrics page.

**Logs:**
I attached the log file since there is so much content there.
[homebridge.log](https://github.com/lstrojny/homebridge-prometheus-exporter/files/9970774/homebridge.log)

**Plugin Config:**

```json
{
    "bridge": {
        "name": "Homebridge",
        "username": "CC:22:3D:E3:CE:32",
        "port": 51090,
        "pin": "031-45-154"
    },
    "description": "Home Smart Home",
    "platforms": [
        {
            "name": "Config",
            "port": 8581,
            "auth": "form",
            "theme": "red",
            "temp": "/sys/class/thermal/thermal_zone0/temp",
            "tempUnits": "c",
            "lang": "auto",
            "platform": "config"
        },
        {
            "name": "deCONZ",
            "anyOn": true,
            "configuredName": true,
            "effects": true,
            "hosts": [
                "192.168.1.3:8081"
            ],
            "lights": true,
            "nativeHomeKitLights": true,
            "nativeHomeKitSensors": true,
            "resource": true,
            "sensors": true,
            "users": {
                "xxx": "xxx"
            },
            "platform": "Hue"
        },
        {
            "name": "Miele@home",
            "clientID": "xxx",
            "clientSecret": "xxx",
            "language": "de",
            "reconnectEventServerInterval": 60,
            "disableStopActionFor": [
                "Dishwasher"
            ],
            "platform": "Miele@home"
        },
        {
            "name": "Shelly",
            "username": "xxx",
            "password": "xxx",
            "admin": {
                "enabled": true
            },
            "devices": [
                {
                    "id": "xxx",
                    "name": "Wohnzimmer Decke",
                    "type": "switch"
                },
                {
                    "id": "xxx",
                    "name": "Arbeitszimmer",
                    "type": "valve",
                    "exclude": false
                }
            ],
            "platform": "Shelly"
        },
        {
            "pin": "031-45-154",
            "debug": false,
            "prefix": "homebridge",
            "port": 36123,
            "refresh_interval": 60,
            "request_timeout": 10,
            "discovery_timeout": 20,
            "platform": "PrometheusExporter"
        }
    ],
    "accessories": [
        {
            "name": "Pihole",
            "auth": "xxx",
            "ssl": false,
            "rejectUnauthorized": true,
            "host": "192.168.1.3",
            "port": 80,
            "baseDirectory": "/admin/",
            "time": 0,
            "reversed": false,
            "logLevel": 1,
            "accessory": "Pihole"
        },
        {
            "accessory": "PhilipsTV",
            "name": "TV Wohnzimmer",
            "ip_address": "192.168.1.124",
            "poll_status_interval": "60",
            "model_year": 2016,
            "has_ssl": true,
            "has_ambilight": false,
            "wol_url": "wol://54:2A:A2:FA:B7:2E",
            "username": "xxx",
            "password": "xxx"
        },
        {
            "accessory": "mi-flower-care",
            "name": "Blumenkasten rechts",
            "deviceId": "C4:7C:8D:6C:C8:AE",
            "interval": 60
        },
        {
            "accessory": "mi-flower-care",
            "name": "Blumenkasten links",
            "deviceId": "C4:7C:8D:6C:CD:6A",
            "interval": 60
        }
    ],
    "disabledPlugins": [
        "homebridge-mieleathome",
        "homebridge-notification",
        "homebridge-mi-flower-care"
    ]
}
```

**Screenshots:**

**Environment:**

* **Plugin Version**: v0.0.3
* **Homebridge Version**: v1.5.1
* **Node.js Version**: v18.12.1
* **NPM Version**: v8.19.2
* **Operating System**: Raspbian

**Comment by lstrojny** [comment]
Thank you for the report, looking into it right now.

**Comment by lstrojny** [comment]
Fixed in the develop branch, rolling a new release now

**Comment by lstrojny** [comment]
Published 0.0.4 to NPM which should fix the issue. Thanks again for the report and good luck trying the new version.

**Comment by frequentsailor** [comment]
Wow that was quick. Thank you very much for the quick fix. It looks better now.
Unfortunately I am running into a different error message in the log. I will open a new issue for that.
